### PR TITLE
koord-descheduler: refactor descheduler pod sorter

### PR DESCRIPTION
**Pull request by eahydra**
Signed-off-by: Joseph <joseph.t.lee@outlook.com>

### Ⅰ. Describe what this PR does

refactor koord-descheduler pod sorter:
1. Implement independent comparison functions according to different dimensions, improve code readability and facilitate code reuse
2. Optimize the implementation of Pod/Node sorting by usage

### V. Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `make test`

### CODE REVIEW

File: pkg/descheduler/framework/plugins/loadaware/utilization_util.go [hunk 1]
```diff
@@ -343,6 +352,27 @@ func evictPods(
 	}
 }
 
+// sortNodesByUsage sorts nodes based on usage according.
```
Comment (jasonliu747) [inline pkg/descheduler/framework/plugins/loadaware/utilization_util.go:355]: according -> accordingly

File: pkg/descheduler/framework/plugins/loadaware/utilization_util.go [hunk 2]
```diff
@@ -389,7 +419,7 @@ func classifyPods(pods []*corev1.Pod, filter func(pod *corev1.Pod) bool) ([]*cor
 	return nonRemovablePods, removablePods
 }
 
-func calcAverageResourceUsagePercent(nodeUsages []*NodeUsage) ResourceThresholds {
+func calcAverageResourceUsagePercent(nodeUsages map[string]*NodeUsage) ResourceThresholds {
```
Comment (jasonliu747) [inline pkg/descheduler/framework/plugins/loadaware/utilization_util.go:422]: slice 改成 map 应该不会影响下面的逻辑，double check一下

File: pkg/descheduler/framework/plugins/loadaware/utilization_util.go (hunk 1 above)
Comment (eahydra) [reply pkg/descheduler/framework/plugins/loadaware/utilization_util.go]: updated

**Review by jasonliu747 — COMMENTED** [review]
/lgtm

**Comment by eahydra** [comment]
updated to resolve conflicts

**Review by FillZpp — APPROVED** [review]
/lgtm

**Pull request merged**